test/test_helper.rb

```diff
@@ -1,45 +1,78 @@
+ENV['RAILS_ENV'] = 'test'
+
 require 'codeclimate-test-reporter'
-CodeClimate::TestReporter.start
+require 'simplecov'
+
+class SimpleCov::Formatter::MergedFormatter
+  def format(result)
+    SimpleCov::Formatter::HTMLFormatter.new.format(result)
+    CodeClimate::TestReporter::Formatter.new.format(result)
+ end
+end
 
-ENV["RAILS_ENV"] = "test"
+if ENV['CODECLIMATE_REPO_TOKEN']
+  CodeClimate::TestReporter.start
+  SimpleCov.formatter = SimpleCov::Formatter::MergedFormatter
+else
+  SimpleCov.start do
+    add_filter '/test/'
+    add_filter '/config/'
+    add_filter '/vendor/'
+
+    add_group 'Controllers', 'app/controllers'
+    add_group 'Models', 'app/models'
+    add_group 'Services', 'app/services'
+    add_group 'Serializers', 'app/serializers'
+    add_group 'Libs', 'lib/'
+  end
+  SimpleCov.formatter = SimpleCov::Formatter::HTMLFormatter
+end
```

Comment:
The coverage stuff is getting a bit hefty with these changes, would it be possible to break it into a separate file?